### PR TITLE
Update API for personalized results

### LINKED ISSUE #101: Instellen mislukt, wordt opnieuw geprobeerd
Thanks for the great Frank Energy plugin! I have installed the plugin using the documentatie. Everything went fine, exempt for fetching the data. I am stuck with the above error and no devices or entities are installed. I enabled the log and it says:

`2024-02-14 12:22:07.977 DEBUG (MainThread) [custom_components.frank_energie.coordinator] Fetching Frank Energie data
2024-02-14 12:22:08.384 INFO (MainThread) [custom_components.frank_energie.coordinator] No gas prices found for user, falling back to public prices
2024-02-14 12:22:08.384 INFO (MainThread) [custom_components.frank_energie.coordinator] No electricity prices found for user, falling back to public prices
2024-02-14 12:22:08.473 DEBUG (MainThread) [custom_components.frank_energie.coordinator] Finished fetching Frank Energie coordinator data in 0.497 seconds (success: False)`

Maybe it is a password error? Where is the username and password stored?

### PR DESCRIPTION
**Pull request by DCSBL**
Fixes #101

### CODE REVIEW

**Comment by AbeltjeNL** [comment]
Works on my machine. Thanx!

**Comment by Blupfis** [comment]
Just wanted to drop a quick note to say that the code is working flawlessly, running without any errors for the past ±24 hours. Thanks!

**Comment by tbrasser** [comment]
Works a treat!

File: custom_components/frank_energie/__init__.py [hunk 1]
```diff
@@ -20,13 +20,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
...
+        hass.config_entries.async_update_entry(entry, title=title)
+
     # Initialise the coordinator and save it as domain-data
     api = FrankEnergie(
         clientsession=async_get_clientsession(hass),
         auth_token=entry.data.get(CONF_ACCESS_TOKEN, None),
         refresh_token=entry.data.get(CONF_TOKEN, None),
     )
-    frank_coordinator = FrankEnergieCoordinator(hass, entry, api)
+    frank_coordinator = FrankEnergieCoordinator(hass, entry, api, entry.data.get("site_reference"))
```
Comment (bajansen) [inline custom_components/frank_energie/__init__.py]: Is een aparte `site_reference` parameter nodig als dit altijd deel is van het `entry`-object?

File: custom_components/frank_energie/__init__.py [hunk 2]
```diff
@@ -20,13 +20,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
...
+        )
+        me = await api.me()
+
+        # filter out all sites that are not in delivery
+        me.deliverySites = [site for site in me.deliverySites if site.status == "IN_DELIVERY"]
+
+        if len(me.deliverySites) == 0:
+            raise Exception("No suitable sites found for this account")
+
+        site = me.deliverySites[0]
```
Comment (bajansen) [inline custom_components/frank_energie/__init__.py:38]: Voor nu prima, maar inderdaad netter om dit aan de gebruiker te laten (https://github.com/bajansen/home-assistant-frank_energie/issues/104)

File: custom_components/frank_energie/__init__.py (hunk 1 above)
Comment (DCSBL) [reply custom_components/frank_energie/__init__.py]: Oh, nope. Good catch

Comment (DCSBL) [reply custom_components/frank_energie/__init__.py]: Fixed

**Comment by Knoeier66** [comment]
Great work.
Had to remove the integration, re-downloaded, re-install, restart and working again.
Core
2024.3.0
Supervisor
2024.02.1
Operating System
12.0
Frontend
20240306.0

**Pull request merged**